src/app/clusters/ota-requestor/OTARequestor.cpp

```diff
@@ -491,28 +491,39 @@ CHIP_ERROR OTARequestor::SendQueryImageRequest(OperationalDeviceProxy & devicePr
     constexpr OTADownloadProtocol kProtocolsSupported[] = { OTADownloadProtocol::kBDXSynchronous };
     QueryImage::Type args;
 
-    VerifyOrReturnError(Basic::Attributes::VendorID::Get(kRootEndpointId, &args.vendorId) == EMBER_ZCL_STATUS_SUCCESS,
-                        CHIP_ERROR_READ_FAILED);
+    uint16_t vendorId;
+    VerifyOrReturnError(ConfigurationMgr().GetVendorId(vendorId) == CHIP_NO_ERROR, CHIP_ERROR_READ_FAILED);
```

Comment:
Is there a reason to not propagate out the actual failure code?  i.e. to do:
```
ReturnErrorOnFailure(ConfigurationMgr().GetVendorId(vendorId));
```
If there is, it might be worth documenting the reason so someone doesn't "clean this up" later.  And if not, maybe do that?

Similar for the other getters in this function and elsewhere in this PR.

Reply:
No particular reason. I'll return actual failure code.